### PR TITLE
Add requireAuth HOC to manage page accessing.

### PR DESCRIPTION
**Pull request by pink-bunny**

### CODE REVIEW

File: src/hoc/requireAuth.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { Redirect } from 'react-router-dom';
+
+import { isUserLoggedIn as getUserLoggedIn } from '../../state/session/selectors';
+
+export default function requireAuth(WrappedComponent) {
+  function Authentication({ isUserLoggedIn }) {
+    if (!isUserLoggedIn) {
+      return <Redirect to="/" />;
+    }
+    return <WrappedComponent />;
```
Comment (plzen) [inline src/hoc/requireAuth.js]: Pass props to WrappedComponent

File: src/hoc/requireAuth.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+import { isUserLoggedIn as getUserLoggedIn } from '../../state/session/selectors';
+
+export default function requireAuth(WrappedComponent) {
+  function Authentication({ isUserLoggedIn }) {
```
Comment (plzen) [inline src/hoc/requireAuth.js]: Use arrow functions as for other cases

File: src/hoc/requireAuth.js (hunk 1 above)
Comment (pink-bunny) [reply src/hoc/requireAuth.js]: Done.

File: src/hoc/requireAuth.js (hunk 2 above)
Comment (pink-bunny) [reply src/hoc/requireAuth.js]: Done.

**Pull request merged**